redis/RedisLexer.g4

```diff
@@ -0,0 +1,234 @@
+BLMPOP     : 'BLMPOP';
+LPOP       : 'LPOP';
+BLPOP      : 'BLPOP';
+RPOP       : 'RPOP';
+BRPOP      : 'BRPOP';
+RPOPLPUSH  : 'RPOPLPUSH';
+BRPOPLPUSH : 'BRPOPLPUSH';
+LINDEX     : 'LINDEX';
+LINSERT    : 'LINSERT';
+LLEN       : 'LLEN';
+LPOS       : 'LPOS';
+LPUSH      : 'LPUSH';
+LPUSHX     : 'LPUSHX';
+RPUSH      : 'RPUSH';
+RPUSHX     : 'RPUSHX';
+LRANGE     : 'LRANGE';
+LREM       : 'LREM';
+LSET       : 'LSET';
+LTRIM      : 'LTRIM';
+
+// Set keywords
+
+SADD        : 'SADD';
+SCARD       : 'SCARD';
+SDIFF       : 'SDIFF';
+SDIFFSTORE  : 'SDIFFSTORE';
+SINTER      : 'SINTER';
+SINTERCARD  : 'SINTERCARD';
+SINTERSTORE : 'SINTERSTORE';
+SISMEMBER   : 'SISMEMBER';
+SMISMEMBER  : 'SMISMEMBER';
+SMEMBERS    : 'SMEMBERS';
+SMOVE       : 'SMOVE';
+SPOP        : 'SPOP';
+SRANDMEMBER : 'SRANDMEMBER';
+SREM        : 'SREM';
+SSCAN       : 'SSCAN';
+SUNION      : 'SUNION';
+SUNIONSTORE : 'SUNIONSTORE';
+
+// Sorted set keywords
+
+ZMPOP            : 'ZMPOP';
+BZMPOP           : 'BZMPOP';
+ZPOPMAX          : 'ZPOPMAX';
+BZPOPMAX         : 'BZPOPMAX';
+ZPOPMIN          : 'ZPOPMIN';
+BZPOPMIN         : 'BZPOPMIN';
+ZADD             : 'ZADD';
+ZCARD            : 'ZCARD';
+ZCOUNT           : 'ZCOUNT';
+ZDIFF            : 'ZDIFF';
+ZDIFFSTORE       : 'ZDIFFSTORE';
+ZINCRBY          : 'ZINCRBY';
+ZINTER           : 'ZINTER';
+ZINTERCARD       : 'ZINTERCARD';
+ZINTERSTORE      : 'ZINTERSTORE';
+ZLEXCOUNT        : 'ZLEXCOUNT';
+ZSCORE           : 'ZSCORE';
+ZMSCORE          : 'ZMSCORE';
+ZRANDMEMBER      : 'ZRANDMEMBER';
+ZRANGE           : 'ZRANGE';
+ZRANGEBYLEX      : 'ZRANGEBYLEX';
+ZRANGEBYSCORE    : 'ZRANGEBYSCORE';
+ZRANGESTORE      : 'ZRANGESTORE';
+ZRANK            : 'ZRANK';
+ZREM             : 'ZREM';
+ZREMRANGEBYLEX   : 'ZREMRANGEBYLEX';
+ZREMRANGEBYRANK  : 'ZREMRANGEBYRANK';
+ZREMRANGEBYSCORE : 'ZREMRANGEBYSCORE';
+ZREVRANGE        : 'ZREVRANGE';
+ZREVRANGEBYLEX   : 'ZREVRANGEBYLEX';
+ZREVRANGEBYSCORE : 'ZREVRANGEBYSCORE';
+ZREVRANK         : 'ZREVRANK';
+ZSCAN            : 'ZSCAN';
+ZUNION           : 'ZUNION';
+ZUNIONSTORE      : 'ZUNIONSTORE';
+
+// Hash keywords
+
+HDEL         : 'HDEL';
+HEXISTS      : 'HEXISTS';
+HEXPIRE      : 'HEXPIRE';
+HPEXPIRE     : 'HPEXPIRE';
+HEXPIREAT    : 'HEXPIREAT';
+HPEXPIREAT   : 'HPEXPIREAT';
+HEXPIRETIME  : 'HEXPIRETIME';
+HPEXPIRETIME : 'HPEXPIRETIME';
+HGET         : 'HGET';
+HMGET        : 'HMGET';
+HGETALL      : 'HGETALL';
+HINCRBY      : 'HINCRBY';
+HKEYS        : 'HKEYS';
+HLEN         : 'HLEN';
+HSET         : 'HSET';
+HMSET        : 'HMSET';
+HSETNX       : 'HSETNX';
+HPERSIST     : 'HPERSIST';
+HTTL         : 'HTTL';
+HPTTL        : 'HPTTL';
+HRANDFIELD   : 'HRANDFIELD';
+HSCAN        : 'HSCAN';
+HSTRLEN      : 'HSTRLEN';
+HVALS        : 'HVALS';
+
+// Constructors symbols
+
+SINGLE_QUOTE : '\'';
+DOUBLE_QUOTE : '"';
+
+// Literal Primitives
+
+fragment DECIMAL_DIGIT: [0-9];
+
+POSITIVE_DECIMAL_LITERAL : DECIMAL_DIGIT+;
+DECIMAL_LITERAL          : [-]? POSITIVE_DECIMAL_LITERAL;
```

Comment:
`[-]` is not a valid range set. If the intent is to recognize the character '-' then use `'-'`. Likewise below, the character set `[+ | -]` is not valid. I think you mean `('+' | '-')` or `[+\-]`.